pkg/segment/structs/segstructs.go

```diff
@@ -543,9 +555,28 @@ func (qa *QueryAggregators) hasLetColumnsRequest() bool {
 			qa.OutputTransforms.LetColumns.FormatResults != nil || qa.OutputTransforms.LetColumns.EventCountRequest != nil)
 }
 
+func (qa *QueryAggregators) hasHeadBlock() bool {
+	if qa == nil {
+		return false
+	}
+	if qa.OutputTransforms == nil {
+		return false
+	}
+	if qa.OutputTransforms.HeadRequest == nil {
+		return false
+	}
+	if qa.OutputTransforms.HeadRequest.BoolExpr != nil {
+		return true
+	}
+	if qa.OutputTransforms.HeadRequest.MaxRows > qa.OutputTransforms.HeadRequest.RowsAdded {
+		return true
+	}
+	return false
+}
```

Comment:
I think he needs those conditions because `BoolExpr` can be nil but `MaxRows` can be greater than zero.
As you can just given `head 20` or just `head`.

Then BoolExpr` will be nil.

Reply:
You're right; I think I'm just confused what this function is supposed to do. With the name `hasHeadBlock()`, I would expect it to return `true` any time `OutputTransforms.HeadRequest` is not nil. I guess these other checks could be useful for ensuring the non-nil head block is valid though; in that case, why does it matter that `MaxRows > RowsAdded`?

Reply:
This check was present already, my best guess is that once we have processed the required number of rows we would not want this aggregation to be processed even though it is present cause we have added the required number of rows.